hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/compact/RunCompactionActionExecutor.java

```diff
@@ -70,8 +73,19 @@ public HoodieWriteMetadata<HoodieData<WriteStatus>> execute() {
       HoodieCompactionPlan compactionPlan =
           CompactionUtils.getCompactionPlan(table.getMetaClient(), instantTime);
 
+      // try to load internalSchema to support schema Evolution
+      HoodieWriteConfig configCopy = config;
+      Pair<Option<String>, Option<String>> schemaPair = InternalSchemaCache
```

Comment:
What is the purpose we re-assign the config to `configCopy` and modify it directly ? I mean, you should either modify the `config` directly or copy the whole config to `configCopy` !

Reply:
origin config will not be modfy,  when schema evolution happen，we will copy whole config to configCopy and modfiy the copy config； otherwize nothing will happen
maybe i miss something

Reply:
The `configCopy` and `config` reference the same Java object ;)